src/Orc.Notifications/Orc.Notifications.Shared/Services/NotificationService.cs

```diff
@@ -121,46 +125,50 @@ public void ShowNotification(INotification notification)
 
             EnsureMainWindow();
 
-            _dispatcherService.BeginInvoke(() =>
+            if (notification.Priority > NotificationPriority.Normal || Application.Current.Windows.OfType<Window>().FirstOrDefault(window => window.IsActive) != null)
```

Comment:
Please store the window value inside a variable. Then we can debug the result and we can analyze afterwards why a specific notification was (not) shown.

so:

```
var hasActiveWindows = HasActiveWindows();
if (!hasActiveWindows && notification.Priority <= NotificationPriority.Normal)
{
    Log.DEbug($"Not showing notification '{notification}' since priority is '{notification.Priority}' and app has no active windows."
    return;
}
```
Also, please add a virtual method to determine the active window:

```
protected virtual bool HasActiveWindows()
{
    return Application.Current.Windows.OfType<Window>().FirstOrDefault(window => window.IsActive) != null;
}
```

This way users can override the behavior.

Reply:
I will do, but this change has a performance issue. We only need to look for the active window if priority <= NotificationPriority.Normal, not look for the windows first.